build/replaceRef.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main implements a program that replaces ref with contents in the given json files
+package main
+
+import (
+	"log"
+	"os"
+	"strings"
+)
+
+func main() {
+	if len(os.Args) != 4 {
+		log.Println("Usage: replaceRef <jsonFilePath> <ref> <contents>")
+		return
+	}
+
+	jsonFilePath := os.Args[1]
+	ref := os.Args[2]
+	contents := os.Args[3]
+
+	processJSON(jsonFilePath, ref, contents)
```

Comment:
I totally realise how what I said would be confusing, but my comment was more to replace the entire bash file with a Go program, not just this bit 😃

But maybe this is easier?

Reply:
got it👍🏻, will covert the bash file with Go 🙂

Reply:
I've completed a Go script to convert from json to yaml file - https://gist.github.com/Kalaiselvi84/c0487628acdfda6f8bb62047b8e88036